treeherder/perf/autoclassify.py

```diff
@@ -0,0 +1,84 @@
+import json
+from datetime import timedelta
+
+from treeherder.etl.common import make_request
+from treeherder.model.derived import JobsModel
+from treeherder.perf.models import PerformanceAlertSummary
+from treeherder.webapp.api.utils import UrlQueryFilter
+
+
+def get_result_set(repository, result_set_id):
+    filter = UrlQueryFilter({"id": result_set_id})
+    full = filter.pop("full", "true").lower() == "true"
+    with JobsModel(repository) as jm:
+        result = jm.get_result_set_list(0, 1, full, filter.conditions)[0]
+    return result
+
+
+def is_downstream(summary):
+    """Returns a Bool of whether or not the summary given is upstream"""
+    for i in range(int(summary.result_set_id), int(summary.prev_result_set_id), -1):
+        try:
+            result_set = get_result_set(summary.repository.name, i)
+        except IndexError:
+            return False
+        if "merge" in result_set["comments"].lower():
```

Comment:
Merge could occur out of the context of an actual merge. I'd probably look for this text explicitly: "merge XXXXX to YYYYY", you can do this via a regular expression.